care/facility/api/viewsets/patient.py

```diff
@@ -147,8 +147,25 @@ def filter_by_category(self, queryset, name, value):
         field_name="last_consultation__symptoms_onset_date"
     )
     last_consultation_admitted_bed_type_list = MultiSelectFilter(
-        field_name="last_consultation__current_bed__bed__bed_type"
+        field_name="last_consultation__current_bed__bed__bed_type",
+        method="filter_by_bed_type",
     )
+
+    def filter_by_bed_type(self, queryset, name, value):
+        values = value.split(",")
+        union_queryset = PatientRegistration.objects.none()
+        for val in values:
+            val = val.strip()
+            if val == "None":
+                union_queryset |= queryset.filter(
+                    last_consultation__current_bed__isnull=True
+                )
+            else:
+                union_queryset |= queryset.filter(
+                    last_consultation__current_bed__bed__bed_type=val
+                )
+        return union_queryset
```

Comment:
Use a Q function to write this filter, using an | with the base queryset can make the query really complicated.

Reply:
okay!

Reply:
@vigneshhari , made changes please review the update code!